src/System.IO.Pipelines/src/System/IO/Pipelines/PipeOptions.cs

```diff
@@ -30,17 +30,25 @@ public class PipeOptions
             MemoryPool<byte> pool = null,
             PipeScheduler readerScheduler = null,
             PipeScheduler writerScheduler = null,
-            long pauseWriterThreshold = DefaultPauseWriterThreshold,
-            long resumeWriterThreshold = DefaultResumeWriterThreshold,
-            int minimumSegmentSize = DefaultMinimumSegmentSize,
+            long pauseWriterThreshold = -1,
+            long resumeWriterThreshold = -1,
+            int minimumSegmentSize = -1,
             bool useSynchronizationContext = true)
         {
-            if (pauseWriterThreshold < 0)
+            if (pauseWriterThreshold == -1)
+            {
+                pauseWriterThreshold = DefaultPauseWriterThreshold;
+            }
+            else if (pauseWriterThreshold < 0)
             {
                 ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.pauseWriterThreshold);
             }
 
-            if (resumeWriterThreshold < 0 && resumeWriterThreshold > pauseWriterThreshold)
```

Comment:
I fixed this bug, which means exceptions will start to be thrown in cases where they weren't before.  I assume that's desirable.